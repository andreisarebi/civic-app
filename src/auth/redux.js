export const AUTH_NAMESPACE = 'auth';
export const formTypes = {
  INITIAL: 'initial',
  LOGIN: 'login',
  SIGN_UP: 'signUp',
};

// Actions
export const AuthActionType = {
  EmailLoginRequest: 'civicApp/auth/emailLoginRequest',
  RegisterRequest: 'civicApp/auth/registerRequest',
  FacebookLoginRequest: 'civicApp/auth/facebookLoginRequest',
  GoogleLoginRequest: 'civicApp/auth/googleLoginRequest',
  LoginSuccess: 'civicApp/auth/emailLoginSuccess',
  LogOutRequest: 'civicApp/auth/logOutRequest',
  LogOutSuccess: 'civicApp/auth/logOutSuccess',
  AuthFailure: 'civicApp/auth/failure',
  SwitchFormType: 'civicApp/auth/switchFormType',
  UpdateEmail: 'civicApp/auth/updateEmail',
  UpdatePassword: 'civicApp/auth/updatePassword',
  UpdateDuplicatePassword: 'civicApp/auth/updateDuplicatePassword',
  ShowErrors: 'civicApp/auth/showErrors',
};

// Action Creators
export const emailLogin = () => ({ type: AuthActionType.EmailLoginRequest });

export const register = () => ({ type: AuthActionType.RegisterRequest });

export const facebookLogin = () => ({ type: AuthActionType.FacebookLoginRequest });

export const googleLogin = () => ({ type: AuthActionType.GoogleLoginRequest });

// Action Creators
export const logIn = (email, password) => ({
  type: AuthActionType.LoginRequest,
  payload: { email, password },
});

export const loginSuccess = user => ({
  type: AuthActionType.LoginSuccess,
  payload: user,
});

export const logOut = () => ({ type: AuthActionType.LogOutRequest });

export const logOutSuccess = () => ({ type: AuthActionType.LogOutSuccess });

export const authFailure = error => ({
  type: AuthActionType.AuthFailure,
  payload: error,
});

/*
 * @param formType: member of formTypes object;
 */
export const switchFormType = formType => ({
  type: AuthActionType.SwitchFormType,
  payload: formType,
});

/*
 * @param email: string;
 */
export const updateEmail = email => ({
  type: AuthActionType.UpdateEmail,
  payload: email,
});

/*
 * @param password: string;
 */
export const updatePassword = password => ({
  type: AuthActionType.UpdatePassword,
  payload: password,
});

/*
 * @param password: string;
 */
export const updateDuplicatePassword = password => ({
  type: AuthActionType.UpdateDuplicatePassword,
  payload: password,
});

/*
 * @param shouldShow: boolean;
 */
export const showErrors = shouldShow => ({
  type: AuthActionType.ShowErrors,
  payload: shouldShow,
});

/*
 * When a user is logged in, we will store:
 * id, email, ...
 */
export const initialState = {
  user: null,
  formType: formTypes.INITIAL,
  email: '',
  password: '',
  duplicatePassword: '',
  showErrors: false,
};

const authReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case AuthActionType.SwitchFormType:
      return {
        ...state,
        formType: action.payload,
        showErrors: false,
      };
    case AuthActionType.UpdateEmail:
      return {
        ...state,
        email: action.payload,
        showErrors: false,
      };
    case AuthActionType.UpdatePassword:
      return {
        ...state,
        password: action.payload,
        showErrors: false,
      };
    case AuthActionType.UpdateDuplicatePassword:
      return {
        ...state,
        duplicatePassword: action.payload,
        showErrors: false,
      };
    case AuthActionType.ShowErrors:
      return {
        ...state,
        showErrors: action.payload,
      };
    case AuthActionType.LoginSuccess:
      return {
        ...state,
        user: { id: action.payload.uid, email: action.payload.email },
      };
    case AuthActionType.LogOutSuccess:
      return {
        ...state,
        user: null,
      };
    default:
      return state;
  }
};

export default authReducer;
