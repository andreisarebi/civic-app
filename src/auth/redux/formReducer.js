export const FORM_NAMESPACE = 'form';

export const formTypes = {
  INITIAL: 'initial',
  LOGIN: 'login',
  SIGN_UP: 'signUp',
};

// Actions
export const AuthFormActionType = {
  SwitchFormType: 'civicApp/auth/switchFormType',
  UpdateEmail: 'civicApp/auth/updateEmail',
  UpdatePassword: 'civicApp/auth/updatePassword',
  UpdateDuplicatePassword: 'civicApp/auth/updateDuplicatePassword',
  UpdateErrorMessage: 'civicApp/auth/updateErrorMessage',
};

// Action Creators
export const switchFormType = formType => ({
  type: AuthFormActionType.SwitchFormType,
  payload: formType,
});

export const updateEmail = email => ({
  type: AuthFormActionType.UpdateEmail,
  payload: email,
});

export const updatePassword = password => ({
  type: AuthFormActionType.UpdatePassword,
  payload: password,
});

export const updateDuplicatePassword = password => ({
  type: AuthFormActionType.UpdateDuplicatePassword,
  payload: password,
});

// TODO: errors and maybe loading
const formReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case AuthFormActionType.SwitchFormType:
      return {
        ...state,
        formType: action.payload,
      };
    case AuthFormActionType.UpdateEmail:
      return {
        ...state,
        email: action.payload,
      };
    case AuthFormActionType.UpdatePassword:
      return {
        ...state,
        password: action.payload,
      };
    case AuthFormActionType.UpdateDuplicatePassword:
      return {
        ...state,
        duplicatePassword: action.payload,
      };
    default:
      return state;
  }
};

const initialState = {
  formType: formTypes.INITIAL,
  email: '',
  password: '',
  duplicatePassword: '',
  errorMessage: '',
};

export default formReducer;
