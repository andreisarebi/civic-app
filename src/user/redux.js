export const USER_NAMESPACE = 'user';

// Actions
export const UserActionType = {
  Request: 'civicApp/user/REQUEST',
  RequestSuccess: 'civicApp/user/REQUEST_SUCCESS',
};

// Action creators
export const userFetchSuccess = ({ responses, favorites, district, user }) => ({
  type: UserActionType.RequestSuccess,
  payload: { responses, favorites, district, user },
});

export const loadUser = (force = false) => ({
  type: UserActionType.Request,
  payload: force
});

export const initialState = {
  district: null,
};

const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case UserActionType.RequestSuccess:
      return {
        ...state,
        district: action.payload.district,
      };
    default:
      return state;
  }
}

export default reducer;
