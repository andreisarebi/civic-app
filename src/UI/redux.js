import { AuthUserActionType } from '../auth/redux/userReducer';
import { parseErrorResponse } from '../auth/api';
// selectors
export const getIsSmallScreen = state => state[UI_NAMESPACE].isSmallScreen;
export const getNetworkError = state => state[UI_NAMESPACE].error;
export const getIsLoading = state => state[UI_NAMESPACE].isLoading;
/*
 * Parses error codes returned from Firebase
 */
export const getErrorMessage = state => {
  const error = getNetworkError(state);
  return parseErrorResponse(error);
};

// actions
export const setIsSmallScreen = () => ({
  type: SET_IS_SMALL_SCREEN,
});

// action types
export const SET_IS_SMALL_SCREEN = 'civic-app/ui/setIsSmallScreen';

// reducer
export const UI_NAMESPACE = 'ui';

export const initialState = {
  isSmallScreen: false,
  isLoading: false,
  error: { code: '', message: '' },
};

const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case AuthUserActionType.EmailLoginRequest:
    case AuthUserActionType.FacebookLoginRequest:
    case AuthUserActionType.GoogleLoginRequest:
    case AuthUserActionType.RegisterRequest:
    case AuthUserActionType.LogOutRequest:
      return {
        ...state,
        loading: true,
        error: initialState.error,
      };
    case AuthUserActionType.LoginSuccess:
    case AuthUserActionType.RegisterSuccess:
    case AuthUserActionType.LogOutSuccess:
      return {
        ...state,
        loading: false,
      };
    case AuthUserActionType.AuthFailure:
      return {
        ...state,
        error: action.payload,
        loading: false,
      };
    case SET_IS_SMALL_SCREEN:
      return {
        ...state,
        isSmallScreen: true,
      };
    default:
      return state;
  }
};

export default reducer;
