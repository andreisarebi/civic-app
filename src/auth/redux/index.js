import { combineReducers } from 'redux';
import userReducer, { USER_NAMESPACE } from './userReducer';
import formReducer, { FORM_NAMESPACE } from './formReducer';

export const AUTH_NAMESPACE = 'auth';

const authReducer = combineReducers({
  [USER_NAMESPACE]: userReducer,
  [FORM_NAMESPACE]: formReducer,
});

export default authReducer;
