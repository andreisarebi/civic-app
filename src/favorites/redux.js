import { Category } from './models';
import { UserActionType } from '../user/redux';

// Selectors
export const getUserFavorites = (state) => ({
  [Category.Candidates]: Array.from(getFavoritesForCategory(state, Category.Candidates)),
  [Category.Events]: Array.from(getFavoritesForCategory(state, Category.Events)),
});

export const getIsFavorite = (state, id, category) => (
  getFavoritesForCategory(state, category).has(id)
);

export const getFavoritesForCategory = (state, category) => (
  state[FAVORITES_NAMESPACE][category]
);

export const FAVORITES_NAMESPACE = 'favorites';

// Action Creators
export const favoritesRequestSuccess = favorites => ({
  type: FavoritesActionType.RequestSuccess,
  payload: favorites,
});

export const toggleFavorite = (id, category) => ({
  type: FavoritesActionType.Toggle,
  payload: { id, category },
});

export const addOrRemoveFavorite = (id, category, shouldAdd) => (
  shouldAdd ? addFavorite(id, category) : removeFavorite(id, category)
);

const addFavorite = (id, category) => ({
  type: FavoritesActionType.Add,
  payload: { id, category }
});

const removeFavorite = (id, category) => ({
  type: FavoritesActionType.Remove,
  payload: { id, category },
});

export const FavoritesActionType = {
  Toggle: 'civicApp/favorites/toggle',
  Add: 'civicApp/favorites/add',
  Remove: 'civicApp/favorites/remove',
};

// Reducer

const reducer = (state = initialState, action) => {
  let newState;
  switch (action.type) {
    case UserActionType.RequestSuccess:
      return action.payload.favorites ? {
        [Category.Candidates]: new Set(action.payload.favorites[Category.Candidates]),
        [Category.Events]: new Set(action.payload.favorites[Category.Events]),
      } : state;
    case FavoritesActionType.Add:
      return {
        ...state,
        [action.payload.category]: (new Set(state[action.payload.category])).add(action.payload.id),
      };
    case FavoritesActionType.Remove:
      newState = new Set(state[action.payload.category]);
      newState.delete(action.payload.id);
      return {
        ...state,
        [action.payload.category]: newState,
      };
    default:
      return state;
  }
};

const initialState = {
  [Category.Candidates]: new Set(),
  [Category.Events]: new Set(),
};

export default reducer;
