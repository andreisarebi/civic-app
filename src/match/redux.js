import { combineReducers } from 'redux';
import { UserActionType } from '../user/sagas';

// Selectors
export const MATCH_NAMESPACE = 'match';
const SURVEY_NAMESPACE = 'survey';
const POSITIONS_NAMESPACE = 'userPositions';

export const getSurveyQuestions = (state) => (
  state[MATCH_NAMESPACE][SURVEY_NAMESPACE]
);

export const getUserPositions = (state) => (
  state[MATCH_NAMESPACE][POSITIONS_NAMESPACE]
);

// Action Creators
export const loadSurveySuccess = surveyQuestions => ({
  type: MatchActionType.SurveyRequestSuccess,
  payload: surveyQuestions,
});

export const loadSurvey = () => ({
  type: MatchActionType.SurveyRequest,
});

export const MatchActionType = {
  SurveyRequest: 'civicApp/match/survey/REQUEST',
  SurveyRequestSuccess: 'civicApp/match/survey/REQUEST_SUCCESS',
  UserPositionUpdate: 'civicApp/match/positions/UPDATE',
};

// Reducers
const surveyReducer = (state = null, action) => {
  switch (action.type) {
    case MatchActionType.SurveyRequestSuccess:
      return {
        ...state,
        ...action.payload
      };
    default:
      return state;
  }
};

const positionsReducer = (state = {}, action) => {
  switch (action.type) {
    case UserActionType.RequestSuccess:
      return {
        ...state,
        ...action.payload.responses,
      };
    default:
      return state;
  }
};

export default combineReducers({
  [SURVEY_NAMESPACE]: surveyReducer,
  [POSITIONS_NAMESPACE]: positionsReducer,
});
