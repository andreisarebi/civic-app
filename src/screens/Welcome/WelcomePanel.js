import { StyleSheet, Text, View } from 'react-native';
import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-native-elements';
import colors from '../../styles/colors';
import { formTypes } from '../../auth/redux';
import SocialButton from '../../auth/SocialButton';

class WelcomePanel extends React.Component {
  static propTypes = {
    switchFormType: PropTypes.func,
    formType: PropTypes.string,
    navigate: PropTypes.func,
  };

  sharedButtonProps = {
    borderRadius: 3,
    containerViewStyle: styles.buttonContainer,
    fontSize: 18,
  };

  defaultView = (
    <View style={styles.defaultViewContainer}>
      <Button
        {...this.sharedButtonProps}
        color={colors.white}
        backgroundColor={colors.red}
        title="Register"
        onPress={() => this.props.switchFormType(formTypes.SIGN_UP)}
      />
      <Button
        {...this.sharedButtonProps}
        color={colors.black}
        backgroundColor={colors.offWhite}
        title="Sign In"
        onPress={() => this.props.switchFormType(formTypes.LOGIN)}
      />
    </View>
  );

  expandedView = formType => {
    const config = (type => {
      switch (type) {
        case formTypes.INITIAL:
          return {};
        case formTypes.LOGIN:
          return {
            preposition: 'in',
            switchText: 'Don\'t have an account yet? Register',
            otherFormType: formTypes.SIGN_UP,
          };
        case formTypes.SIGN_UP:
          return {
            preposition: 'up',
            switchText: 'Have an account? Sign in',
            otherFormType: formTypes.LOGIN,
          };
        default:
          return {};
      }
    })(formType);

    return (
      <View style={styles.expandedViewContainer}>
        <SocialButton
          type="google"
          title={`Sign ${config.preposition} with Google`}
          style={styles.social}
        />
        <SocialButton type="facebook" title="Continue with Facebook" style={styles.social} />
        <Text style={styles.text}>or</Text>
        <SocialButton
          type="email"
          title={`Sign ${config.preposition} with email`}
          style={styles.social}
          onPress={this.goToCredentials}
        />
        <Text
          style={[styles.text, styles.switchText]}
          onPress={() => this.props.switchFormType(config.otherFormType)}
        >
          {config.switchText}
        </Text>
      </View>
    );
  };

  goToCredentials = () => {
    this.props.navigate('Credentials');
  };

  render() {
    return (
      <View style={styles.container}>
        {this.props.formType === formTypes.INITIAL
          ? this.defaultView
          : this.expandedView(this.props.formType)}
      </View>
    );
  }
}

var styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    position: 'absolute',
    bottom: 0,
    backgroundColor: colors.white,
    borderTopLeftRadius: 25,
    borderTopRightRadius: 25,
  },
  buttonContainer: {
    flex: 1,
    marginTop: 20,
    marginBottom: 20,
    borderRadius: 3,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.5,
    shadowRadius: 3,
  },
  defaultViewContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  expandedViewContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  social: {
    marginBottom: 8,
    width: '100%',
    maxWidth: 322,
  },
  text: {
    textAlign: 'center',
    fontSize: 18,
    paddingTop: 10,
    paddingBottom: 18,
  },
  switchText: {
    paddingBottom: 0,
  },
});

export default WelcomePanel;
