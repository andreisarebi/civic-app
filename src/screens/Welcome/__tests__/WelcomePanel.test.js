import { createSnapshot } from '../../../util/tests';
import WelcomePanel from '../WelcomePanel';
import { formTypes } from '../../../auth/redux/formReducer';

describe('WelcomePanel', () => {
  it('renders correctly in initial view', () => {
    const props = {
      formType: formTypes.INITIAL,
      switchFormType: jest.fn(),
      navigate: jest.fn(),
    };
    const snapshot = createSnapshot(WelcomePanel, props);
    expect(snapshot).toMatchSnapshot();
  });

  it('renders correctly in login view', () => {
    const props = {
      formType: formTypes.LOGIN,
      switchFormType: jest.fn(),
      navigate: jest.fn(),
    };
    const snapshot = createSnapshot(WelcomePanel, props);
    expect(snapshot).toMatchSnapshot();
  });
  it('renders correctly in sign up view', () => {
    const props = {
      formType: formTypes.SIGN_UP,
      switchFormType: jest.fn(),
      navigate: jest.fn(),
    };
    const snapshot = createSnapshot(WelcomePanel, props);
    expect(snapshot).toMatchSnapshot();
  });
});
