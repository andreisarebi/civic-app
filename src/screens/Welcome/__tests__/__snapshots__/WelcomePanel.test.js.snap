// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePanel renders correctly in initial view 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderTopLeftRadius": 25,
      "borderTopRightRadius": 25,
      "bottom": 0,
      "flexDirection": "row",
      "position": "absolute",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Button
      backgroundColor="#eb5757"
      borderRadius={3}
      color="#ffffff"
      containerViewStyle={
        Object {
          "borderRadius": 3,
          "flex": 1,
          "marginBottom": 20,
          "marginTop": 20,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 3,
        }
      }
      fontSize={18}
      onPress={[Function]}
      title="Register"
    />
    <Button
      backgroundColor="#f2f2f2"
      borderRadius={3}
      color="#000000"
      containerViewStyle={
        Object {
          "borderRadius": 3,
          "flex": 1,
          "marginBottom": 20,
          "marginTop": 20,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 3,
        }
      }
      fontSize={18}
      onPress={[Function]}
      title="Sign In"
    />
  </Component>
</Component>
`;

exports[`WelcomePanel renders correctly in login view 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderTopLeftRadius": 25,
      "borderTopRightRadius": 25,
      "bottom": 0,
      "flexDirection": "row",
      "position": "absolute",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <SocialButton
      style={
        Object {
          "marginBottom": 8,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Sign in with Google"
      type="google"
    />
    <SocialButton
      style={
        Object {
          "marginBottom": 8,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Continue with Facebook"
      type="facebook"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 18,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      or
    </Text>
    <SocialButton
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Sign in with email"
      type="email"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "fontSize": 18,
            "paddingBottom": 18,
            "paddingTop": 10,
            "textAlign": "center",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      Don't have an account yet? Register
    </Text>
  </Component>
</Component>
`;

exports[`WelcomePanel renders correctly in sign up view 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderTopLeftRadius": 25,
      "borderTopRightRadius": 25,
      "bottom": 0,
      "flexDirection": "row",
      "position": "absolute",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <SocialButton
      style={
        Object {
          "marginBottom": 8,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Sign up with Google"
      type="google"
    />
    <SocialButton
      style={
        Object {
          "marginBottom": 8,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Continue with Facebook"
      type="facebook"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 18,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      or
    </Text>
    <SocialButton
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Sign up with email"
      type="email"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "fontSize": 18,
            "paddingBottom": 18,
            "paddingTop": 10,
            "textAlign": "center",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      Have an account? Sign in
    </Text>
  </Component>
</Component>
`;
